akka-docs/rst/scala/code/docs/persistence/query/PersistenceQueryDocSpec.scala

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package docs.persistence.query
+
+import akka.actor._
+import akka.persistence.query.scaladsl.ReadJournal
+import akka.persistence.{ Recovery, PersistentActor }
+import akka.persistence.query._
+import akka.stream.{ FlowShape, ActorMaterializer }
+import akka.stream.scaladsl.FlowGraph
+import akka.stream.scaladsl.{ Flow, Sink, Source }
+import akka.testkit.AkkaSpec
+import akka.util.Timeout
+import docs.persistence.query.PersistenceQueryDocSpec.TheOneWhoWritesToQueryJournal
+
+import scala.collection.immutable
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+
+object PersistenceQueryDocSpec {
+
+  implicit val timeout = Timeout(3.seconds)
+
+  class NoopReadJournal(sys: ExtendedActorSystem) extends ReadJournal {
+    override def query[T, M](q: Query[T, M], hints: Hint*): Source[T, M] =
+      Source.empty.mapMaterializedValue(_ => null.asInstanceOf[M])
+  }
+
+  //#my-read-journal
+  class MyReadJournal(system: ExtendedActorSystem) extends ReadJournal {
+
+    // TODO from config
+    private val defaulRefreshInterval: FiniteDuration = 3.seconds
+
+    override def query[T, M](q: Query[T, M], hints: Hint*): Source[T, M] =
+      q match {
+        case EventsByTag(tag, offset) ⇒
+          val props = MyEventsByTagPublisher.props(tag, offset, refreshInterval(hints))
+          Source.actorPublisher[EventEnvelope](props)
+            .mapMaterializedValue(_ ⇒ noMaterializedValue)
```

Comment:
Thanks, that's great feedback @scullxbones - we're not completely happy yet with the API, so all kinds of ideas are very welcome.
Initially we thought that the same API should be implementable from Java as well as Scala, but since we have javadsl and scaladsl separate anyway, maybe we can go ahead and make the Scala API something nice (and make Java "as good as possible").

I have created a new ticket to gather feedback, I'll copy your comment there so that it does not get lost => https://github.com/akka/akka/issues/18229